src/cargo/core/compiler/build_context/mod.rs

```diff
@@ -50,7 +52,31 @@ impl<'a, 'cfg> BuildContext<'a, 'cfg> {
         profiles: &'a Profiles,
         extra_compiler_args: HashMap<Unit<'a>, Vec<String>>,
     ) -> CargoResult<BuildContext<'a, 'cfg>> {
-        let rustc = config.rustc(Some(ws))?;
+        let mut rustc = config.load_global_rustc(Some(ws))?;
+
+        if build_config.clippy_override {
+            let tool = config.get_tool("clippy-driver")?;
```

Comment:
How come this is `get_tool("clippy-driver")` vs just `"clippy-driver"`? I think that the usage for rustc's own wrapper was to read configuration of what the wrapper is, but in clippy's case there's no need to configure what the driver is called I think?